crdppf/static/js/Crdppf/map.js

```diff
@@ -375,140 +159,107 @@ function makeMap(mapOptions, labels){
                 panIcons: false
             }),
             new OpenLayers.Control.Navigation(),
-            scalebar            
+            new OpenLayers.Control.ScaleLine({
+                bottomOutUnits:'',
+                bottomInUnits: '',
+                maxWidth: 200
+            }),
+            new OpenLayers.Control.OverviewMap({
+                layers: [
+                    new OpenLayers.Layer.Image(
+                        "overview",
+                        Crdppf.staticImagesDir + Crdppf.keymap,
+                        new OpenLayers.Bounds(Crdppf.mapOverviewExtent),
+                        new OpenLayers.Size(Crdppf.mapOverviewSizeW, Crdppf.mapOverviewSizeH)
+                    )
+                ],
+                size: new OpenLayers.Size(Crdppf.mapOverviewSizeW, Crdppf.mapOverviewSizeH),
+                maximized: true,
+                isSuitableOverview: function() {
+                    return true;
+                },
+                mapOptions: {
+                    projection: new OpenLayers.Projection(Crdppf.mapSRS),
+                    displayProjection: new OpenLayers.Projection(Crdppf.mapSRS),
+                    units: "m",
+                    theme: null
+                }
+            })
         ]
-    });   
+    }),
+    // The map initialization function
+    mapInit : function(){
 
-        
-    // Event registering & Control setting on the Map Object
-    map.events.register("mousemove", map, function(e) {
-                var pixel = new OpenLayers.Pixel(e.xy.x,e.xy.y);
-                var lonlat = map.getLonLatFromPixel(pixel);
-                OpenLayers.Util.getElement(mapOptions.divMousePosition).innerHTML = '<b>' + labels.olCoordinates + ' (ch1903) - Y : ' + Math.round(lonlat.lon) + ' m / X : ' + Math.round(lonlat.lat) + ' m</b>';
-    });
-    
-    // add base layers & selection layers
-    map.addLayers([intersect,select, layer]);
-  
-    // create an overview map control and customize it
-    var overviewMap = new OpenLayers.Control.OverviewMap({
-            layers: [
-                new OpenLayers.Layer.Image(
-                    "overview",
-                    Crdppf.staticImagesDir + Crdppf.keymap,
-                    new OpenLayers.Bounds(Crdppf.mapOverviewExtent),
-                    new OpenLayers.Size(Crdppf.mapOverviewSizeW, Crdppf.mapOverviewSizeH)
-                )
-            ],
-            size: new OpenLayers.Size(Crdppf.mapOverviewSizeW, Crdppf.mapOverviewSizeH),
-            maximized: true,
-            isSuitableOverview: function() {
-                return true;
-            },
-            mapOptions: {
-                projection: new OpenLayers.Projection(Crdppf.mapSRS),
-                displayProjection: new OpenLayers.Projection(Crdppf.mapSRS),
-                units: "m",
-                theme: null
-            }
+        // scope trick
+        var me = this;
```

Comment:
Hmmm... this reminds me something ;-)

Isn't there a way to declare the scope in the `this.map.events.register` in order to avoid `var = me = this;`